types/src/v2/params.rs

```diff
@@ -399,6 +398,16 @@ mod test {
 		assert!(obj.is_ok());
 	}
 
+	#[test]
+	fn params_sequence_borrows() {
+		let params = RpcParams::new(Some(r#"["foo", "bar"]"#));
+		let mut seq = params.sequence();
+
+		assert_eq!(seq.next::<&str>().unwrap(), "foo");
+		assert_eq!(seq.next::<&str>().unwrap(), "bar");
+		assert!(seq.next::<&str>().is_err());
```

Comment:
This is the important test and the reason for adding a separate struct for handling sequence parsing: `seq` borrows the json from `params`, which allows it to produce slices with that lifetime, while `&mut self` borrows are transient.